fe/fe-core/src/main/java/com/starrocks/http/rest/TransactionLoadAction.java

```diff
@@ -184,192 +213,593 @@ public void executeWithoutPassword(BaseRequest request, BaseResponse response) t
+            }
         }
-        if (channelNumStr == null && channelIdStr != null) {
-            throw new DdlException("Must provide channel_num when stream load begin.");
+    }
+
+    /**
+     * Transaction management request handler without channel info (eg. id, num) in request.
+     */
+    private class TransactionWithoutChannelHandler implements TransactionHandler {
+
+        @Override
+        public void handle(BaseRequest request, BaseResponse response) throws UserException {
+            RequestParams rp = RequestParams.from(request);
+            Operation op = rp.getOperation();
+            String dbName = rp.getDbName();
+            String tableName = rp.getTableName();
+            String label = rp.getLabel();
+            Channel channel = rp.getChannel();
+            Validate.isTrue(channel.isNull(), "channel isn't null");
+            LOG.info("Handle transaction without channel info, label: {}, source: {}", label, rp.getSourceType());
+
+            Database db = Optional.ofNullable(GlobalStateMgr.getCurrentState().getDb(dbName))
+                    .orElseThrow(() -> new UserException(String.format("Database[%s] does not exist.", dbName)));
+
+            // 1. handle transaction op
+            TransactionResult result = null;
+            switch (op) {
+                case TXN_BEGIN:
+                    Table table = Optional.ofNullable(db.getTable(tableName))
+                            .orElseThrow(() -> new UserException(
+                                    String.format("Table[%s.%s] does not exist.", dbName, tableName)));
+                    result = handleBeginTransaction(db, table, label, rp.getSourceType(), rp.getTimeoutMillis());
+                    break;
+                case TXN_PREPARE:
+                    result = handlePrepareTransaction(
+                            db, label,
+                            Optional.ofNullable(rp.getBody().getCommittedTablets()).orElse(new ArrayList<>(0)),
+                            Optional.ofNullable(rp.getBody().getFailedTablets()).orElse(new ArrayList<>(0))
+                    );
+                    break;
+                case TXN_COMMIT:
+                    result = handleCommitTransaction(db, label, rp.getTimeoutMillis());
+                    break;
+                case TXN_ROLLBACK:
+                    result = handleRollbackTransaction(
+                            db, label,
+                            Optional.ofNullable(rp.getBody().getFailedTablets()).orElse(new ArrayList<>(0))
+                    );
+                    break;
+                case TXN_LOAD:
+                    break;
+                default:
+                    throw new UserException("Unsupported operation: " + op);
+            }
+
+            if (null != result) {
+                sendResult(request, response, result);
+                return;
+            }
+
+            // 2. redirect transaction op to BE
```

Comment:
why redirect to BE？

Reply:
This is the original design and implementation, and i keep it when do refactoring. Howeaver, it isn't needed for bypass writing scenario. 

My understanding about it is that, the StreamLoad task is actually executed by BE. The FE node is only responsible for request forwarding, and the BE Coordinator is responsible for managing the StreamLoad task. Therefore, requests for transactions will also be forwarded to BE Coordinator, which requests FE to manage the transactions through Thrift RPC latter. Maybe @meegoo could give a better answer.